api/envoy/config/listener/v3/listener.proto

```diff
@@ -126,7 +126,7 @@ message Listener {
   // that is governed by the bind rules of the OS. E.g., multiple listeners can listen on port 0 on
   // Linux as the actual port will be allocated by the OS.
   // Required unless ``api_listener`` or ``listener_specifier`` is populated.
-  core.v3.Address address = 2;
+  core.v3.Address address = 2 [(validate.rules).message = {required: true}];
```

Comment:
This check is removed by this PR https://github.com/envoyproxy/envoy/pull/21132 since it isn't required by API listener. cc @markdroth 

But before we have multiple addresses, we will resolve the address first. It will help us to check the address existed or not
https://github.com/envoyproxy/envoy/blob/07f78c92039028319c0240b582ba7c9233c29508/source/common/network/resolver_impl.cc#L49

After multiple addresses, we call `Network::Utility::protobufAddressSocketType` first on the address, and that method will invoke the panic.

There probably two ways to fix that, one is check the address when create ListenerImpl, second is change `Network::Utility::protobufAddressSocketType` to not panic, instead of raising exception.

Reply:
Sorry, where do multiple addresses come from?

Reply:
Sorry, I mean the feature I added for listener to support multiple addresses, https://github.com/envoyproxy/envoy/issues/11184, this is probably the pr may introduce the problem https://github.com/envoyproxy/envoy/pull/22005

Reply:
Makes sense, probably should validate this early in order to provide a meaningful configuration error. I don't think we provide something super useful in `protobufAddressSocketType`, since the error is a missing address, not a missing socket type.

Reply:
> Makes sense, probably should validate this early in order to provide a meaningful configuration error. I don't think we provide something super useful in `protobufAddressSocketType`, since the error is a missing address, not a missing socket type.

+1

Reply:
I will add check in `addOrUpdateListener()`.